src/dspbase.jl

```diff
@@ -660,8 +660,16 @@ function _conv_td!(out, output_indices, u::AbstractArray{<:Number, N}, v::Abstra
 end
 
 # whether the given axis are to be considered to carry an offset for `conv!` and `conv`
-conv_with_offset(::Base.OneTo) = false
-conv_with_offset(a::Any) = throw(ArgumentError("unsupported axis type $(typeof(a))"))
+conv_axis_with_offset(::Base.OneTo) = false
+conv_axis_with_offset(a::Any) = throw(ArgumentError("unsupported axis type $(typeof(a))"))
+
+function conv_axes_with_offset(as::Tuple...)
+    with_offset = ((map(a -> map(conv_axis_with_offset, a), as)...)...,)
+    if !allequal(with_offset)
+        throw(ArgumentError("cannot mix offset and non-offset axes"))
+    end
+    return first(with_offset)
```

Comment:
Ah, good catch. Prior to this PR, we have
```julia
julia> conv(fill(1), fill(1))
0-dimensional Array{Int64, 0}:
1
```
Though probably not terribly important, it might be a good idea to preserve that. And we don't want to make that a 0-dimensional `OffsetArray`, so this should return `false` for this case.
```suggestion
    return !isempty(withoffset) && first(with_offset)
```